content/blog/2022-05-18-learned-federated-query-optimization.md

```diff
@@ -0,0 +1,178 @@
++++
+title = "A Learned Query Compiler"
+[extra]
+latex = true
+[[extra.authors]]
+name = "Victor Giannakouris"
+link = "https://www.linkedin.com/in/vgian/"
++++
+
+# What was the goal?
+The goal of this project was to create an optimizer for a cross-database query compiler. A cross-database query, also 
+known as federated query, is an SQL query that consists of tables that span across different, individual database 
+systems. The main weakness with the already existing federated query optimization approaches is of the complexity
+of integration, as the existing solutions need to obtain specific knowledge from the externally connected databases
+by writing complex drivers. We propose a solution based on a machine learning model that enables the easy integration of
+any SQL-based system, with less than 10 loc.
+
+## Federated Query Compilation Challenges
+A lot of interesting challenges are arising in cross-database query compilation, due to the heterogeneity of the underlying 
+infrastructure. An interesting challenge that arises is the following: How could the query plan be split to further subqueries and how the execution of these subqueries could be orchestrated across the different data sources in order to utilize the external data systems and take advantage of their powerful implementations (e.g. access methods and join algorithms)? 
```

Comment:
This last sentence gets a little long and hard to follow. Maybe it should be 2–3 separate questions?